wallet-api/src/Wallet/API.hs

```diff
@@ -224,6 +231,13 @@ class MonadError WalletAPIError m => WalletDiagnostics m where
 
 -- | Used by Plutus client to interact with wallet
 class WalletAPI m where
+    {- |
+    How much time to leave for transaction validation. The standard validation
+    interval for transactions submitted by this wallet is @(c, c +
+    defaultValidationTime)@ where @c@ is the current slot.
+    -}
+    validationInterval :: m (Slot, Slot)
```

Comment:
Oh I see, it's because you set the interval indirectly and then pull it out of here. I think if we do what I suggest and just pass it then we can get rid of this, which I think we should.

Reply:
done